src/mem3/src/mem3.erl

```diff
@@ -196,7 +196,7 @@ choose_shards(DbName, Nodes, Options) ->
     Q = mem3_util:q_val(couch_util:get_value(q, Options,
         config:get("cluster", "q", "8"))),
     %% rotate to a random entry in the nodelist for even distribution
-    {A, B} = lists:split(crypto:rand_uniform(1,length(Nodes)+1), Nodes),
+    {A, B} = lists:split(couch_rand:uniform(length(Nodes)+1), Nodes),
```

Comment:
Note that `couch_rand:uniform(N)` has a `[1, N]` range, while `crypto:rand_uniform(1, N)` has `[1, N)` so we'd want to subtract 1 as in `...(couch_rand:uniform(length(Nodes))...`

Reply:
```
(node1@127.0.0.1)2> couch_rand:uniform(2).
2
(node1@127.0.0.1)3> couch_rand:uniform(2).
2
(node1@127.0.0.1)4> couch_rand:uniform(2).
1
(node1@127.0.0.1)5> couch_rand:uniform(2).
1
(node1@127.0.0.1)6> couch_rand:uniform(2).
1

(node1@127.0.0.1)7> crypto:rand_uniform(1,2).
1
(node1@127.0.0.1)8> crypto:rand_uniform(1,2).
1
(node1@127.0.0.1)9> crypto:rand_uniform(1,2).
1
(node1@127.0.0.1)10> crypto:rand_uniform(1,2).
1
(node1@127.0.0.1)11> crypto:rand_uniform(1,2).
1
(node1@127.0.0.1)12> crypto:rand_uniform(1,2).
1
```
whoa that is sneaky as heck, and will hopefully teach me to not make assumptions about the uniformity of word meaning. Thank you for pointing out what would have been a horrific bug.